src/internal/item-container/index.tsx

```diff
@@ -279,11 +291,11 @@ function ItemContainerComponent(
   }
 
   function onBlur() {
-    // When drag- or resize handle loses focus the transition must be discarded with two exceptions:
+    // When drag- or resize handle loses focus the transition must be submitted with two exceptions:
     // 1. If the last interaction is not "keyboard" (the user clicked on another handle issuing a new transition);
     // 2. If the item is borrowed (in that case the focus moves to the acquired item which is expected).
     if (transition && transition.interactionType === "keyboard" && !transition.isBorrowed) {
-      draggableApi.discardTransition();
+      draggableApi.submitTransition();
```

Comment:
Changed this for consistency. Leaving keyboard-moved item by any interaction other than pressing "Esc" should submit its position